herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -1836,6 +1836,9 @@ public ApplyEntryOnRecovery() {
 
         @Override
         public void accept(LogSequenceNumber t, LogEntry u) {
+            if (dbmanager.isStopped()) {
+                throw new RuntimeException("System was requested to stop, aborting recovery at " + t);
```

Comment:
Can we ad a more "speaking" exception? Like ShutdownException extends HerdDBInternalException or something like that?

Reply:
@diegosalvi 
It would make sense to have a specific Exception if downstream code can handle it, in this case the caller code is the CommitLog and it cannot recover in any way from the exception.

In which way a new exception here can add value ?

Reply:
Good point